### ISSUE TITLE
Render <Resource /> from application's props

### ISSUE THREAD

**Issue by taweesoft** [marmelab/admin-on-rest#1009]
I would like to render <Resource /> which depends on application's props

For example, 
```
constructor(props) {
  super(props);
  this.state = {
    resources: []
  };
}

componentDidMount() {
  let self = this;
  setTimeout(() => {
      self.setState({ resources: ['posts'] });
  },3000);
}
```

and render <Admin /> by using this code

```
render() {
     return (
                     <Admin title='Post Admin' restClient={jsonServerRestClient('https://jsonplaceholder.typicode.com')}>
                {
                    this.state.resources.map(resource => {
                        return <Resource name={resource} list={PostList} />;
                    })
                }
            </Admin>
     )

}
```

`PostList` is the same one as tutorial

but it turns out with an error
<img width="568" alt="screen shot 2560-09-07 at 6 23 42 pm" src="https://user-images.githubusercontent.com/8603253/30161397-7b6575aa-93fa-11e7-8793-5aff6fc96f85.png">

I did some investigation in List.js from your lib. I found `state.admin` doesn't have key which name `posts`

I would like to make a flexible admin page which is can adapt to various endpoint.

**Comment by djhi** [comment]
Not currently but it will be possible in `1.3.0`. You'll have to define an authClient though. @fzaninotto, maybe we could allow using a function as child without an authClient ?

**Comment by fzaninotto** [comment]
yes good idea, but not a must have for 1.3.0

**Comment by fzaninotto** [comment]
Fixed by #1091
